### ISSUE TITLE
Data leakage?

### ISSUE THREAD

**Issue by Swordshinehjy** [chrisliu2007/passivation_layer_screening#1]
https://github.com/chrisliu2007/passivation_layer_screening/blob/596d6975519b47ec825ba94f27661821ebc73c39/model/regression_model_evaluation.ipynb#L1932C38-L1932C38

**Comment by chrisliu2007** [comment]
can you specify which part? And, what do you mean by data leakage?

**Comment by Swordshinehjy** [comment]
There's a typical data leakage issue in the provided code. The problem lies in how the MinMaxScaler is fitted and then used to transform the data. The code fits the X_scaler (MinMaxScaler) on the entire X_input dataset (X_scaler.fit(X_input)).  This includes the data that will later be used for testing (X_test).  This is the core of the data leakage problem. Data leakage occurs when information from the test set inadvertently influences the training process. In this case, the MinMaxScaler learns the minimum and maximum values of each feature across the entire dataset, including the test set.  This means the scaling applied to the training data is influenced by the distribution of the test data.  The test data should be completely unseen during training, and its characteristics should not play a role in how the training data is processed.
